spring-integration-core/src/main/java/org/springframework/integration/channel/PartitionedChannel.java

```diff
@@ -65,7 +66,7 @@ public class PartitionedChannel extends AbstractExecutorChannel {
 	 * sent to this channel.
 	 */
 	public PartitionedChannel(int partitionCount) {
-		this(partitionCount, (message) -> message.getHeaders().get(IntegrationMessageHeaderAccessor.CORRELATION_ID));
+		this(partitionCount, (message) -> Objects.requireNonNull(message.getHeaders().get(IntegrationMessageHeaderAccessor.CORRELATION_ID)));
```

Comment:
Sorry for my bad English, but I think I said:

>  put everything after an arrow on a new line

Remember `120` symbols per line rule still applies.